src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -664,14 +664,13 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
             },
             executionId: id
         };
-        edit.replaceNotebookCells(
-            this.notebookDocument.uri,
-            new NotebookRange(this.notebookDocument.cellCount, this.notebookDocument.cellCount),
-            [
-                notebookCellData // TODO generalize to arbitrary languages and cell types
-            ]
-        );
-        await workspace.applyEdit(edit);
+        await chainWithPendingUpdates(this.notebookDocument, (edit) => {
```

Comment:
Cells were getting inserted into the `NotebookDocument` out of order, needed `chainWithPendingUpdates` to enforce edits in the right sequence